lib/bolt/transport/local/shell.rb

```diff
@@ -67,8 +67,9 @@ def handle_sudo_errors(err, pid)
 
         def copy_file(source, dest)
           if source.is_a?(StringIO)
-            File.open("tempfile", "w") { |f| f.write(source.read) }
-            execute(['mv', 'tempfile', dest])
+            File.open(dest, 'w') { |f| f.write(source.read) }
```

Comment:
I think I see why this was written to a tempfile and then copied before. For the case where the less privileged user needs to do something as root. The case I was running in to was the opposite (root running as a different user). Will need to figure out a solution that works for both.